test/fuzz/utility.h

```diff
@@ -159,9 +159,18 @@ inline std::unique_ptr<TestStreamInfo> fromStreamInfo(const test::fuzz::StreamIn
   ON_CALL(*upstream_host, metadata()).WillByDefault(testing::Return(upstream_metadata));
   test_stream_info->setUpstreamInfo(std::make_shared<StreamInfo::UpstreamInfoImpl>());
   test_stream_info->upstreamInfo()->setUpstreamHost(upstream_host);
-  auto address = stream_info.has_address()
-                     ? Envoy::Network::Address::resolveProtoAddress(stream_info.address())
-                     : Network::Utility::resolveUrl("tcp://10.0.0.1:443");
+  auto address =
+      stream_info.has_address()
+          ? (stream_info.address().address_case() ==
+                         envoy::config::core::v3::Address::AddressCase::kEnvoyInternalAddress &&
+                     stream_info.address().envoy_internal_address().address_name_specifier_case() ==
+                         envoy::config::core::v3::EnvoyInternalAddress::AddressNameSpecifierCase::
+                             kServerListenerName
+                 ? std::make_shared<Envoy::Network::Address::EnvoyInternalInstance>(
+                       replaceInvalidHostCharacters(
+                           stream_info.address().envoy_internal_address().server_listener_name()))
+                 : Envoy::Network::Address::resolveProtoAddress(stream_info.address()))
+          : Network::Utility::resolveUrl("tcp://10.0.0.1:443");
```

Comment:
I think the logic is correct, but nested ternary-operators are hard to read/reason about.
Can you please use if-else here?

Reply:
Changed. Is this how you like to see it?